### PR TITLE
Benchmark with Pytorch Profiler easily

### PR DESCRIPTION
**Pull request by bjmsong**
## Motivation

Users can use PyTorch Profiler during benchmark easily with `--profile`. In this way, the trace file will not be too large.

### Usgae Example
```
export SGLANG_TORCH_PROFILER_DIR=/root/sglang/profile_log
python -m sglang.launch_server --model-path ${model_path}
python -m sglang.bench_serving --backend sglang --model ${model_path} --dataset-path {dataset_path} --num-prompts 3 --profile
```

## Checklist

- [ ] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [ ] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: python/sglang/bench_serving.py [hunk 1]
```diff
@@ -869,6 +871,18 @@ async def benchmark(
...
 
+    if profile:
+        print("Starting profiler...")
+        profile_input = RequestFuncInput(model=model_id,
+                                        prompt=test_prompt,
+                                        api_url=base_url + "/start_profile",
+                                        prompt_len=test_prompt_len,
+                                        output_len=test_output_len,
+                                        extra_request_body=extra_request_body)
+        profile_output = await request_func(request_func_input=profile_input)
```
Comment (merrymercy) [inline python/sglang/bench_serving.py]: Do not reuse `request_func`. Many arguments, such as `prompt_len` and `output_len`, are meaningless or incorrect for the `start_profile` endpoint. Can you create a standalone function for this?

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: python/sglang/bench_serving.py (hunk 1 above)
Comment (bjmsong) [reply python/sglang/bench_serving.py]: Thanks for review, the `request_func` function has been replaced with a standalone function.

File: python/sglang/bench_serving.py [hunk 2]
```diff
@@ -147,8 +147,8 @@ async def async_request_openai_completions(
 ) -> RequestFuncOutput:
     api_url = request_func_input.api_url
     assert api_url.endswith(
-        "completions"
-    ), "OpenAI Completions API URL must end with 'completions'."
+        ("completions", "profile")
```
Comment (merrymercy) [inline python/sglang/bench_serving.py]: revert this.

**Review by merrymercy — CHANGES_REQUESTED** [review]
please also fix the lint error https://sgl-project.github.io/references/contributor_guide.html#format-your-code

File: python/sglang/bench_serving.py (hunk 2 above)
Comment (bjmsong) [reply python/sglang/bench_serving.py]: sorry, it has been reverted.

**Pull request merged**